mhkit/tests/Power_TestDevice.m

```diff
@@ -113,6 +113,129 @@ function test_ac_power_three_phase(testCase)
+            fnm = ls(sprintf('testdata/B.3.%i*_im.txt',opt));
+            i_m0 = readmatrix(fnm);
+            fnm = ls(sprintf('testdata/B.3.%i*_um.txt',opt));
+            u_m0 = readmatrix(fnm);
+            testCase.verifyTrue(max(abs( ...
+                (i_m.data(1:100,1)-i_m0(:,1))./i_m0(:,1) ...
+                ))<1e-10,opt);
+            testCase.verifyTrue(max(abs((u_m.data(1:100)-u_m0)./u_m0))<1e-10,...
+                opt);
+        end
+
+        function test_flicker_ufic(testCase)
+            % u_m, i_m, Sr, Un, In, SCR, fg, & fs
+            Sr = 3e6; Un=12e3; In=144; fg=60; fs=50e3;fv=0.5;
+            %1. opt = 0, pure sine waves:
+            opt = 0; idx = randi([1,4],1);
+            fm=20; SCR=20; DeltaI_I = [0 0 0 0];% pure sine wave
+            [i_m,u_m]=gen_test_data(Un,In,fg,fs,fm,fv,DeltaI_I,opt,650);
+            method = 'ZCD'; methodopts = {};
+            [~,freq,alpha_m,u0,u_fic] = flicker_ufic( ...
+                Sr,Un,SCR,fg,u_m,i_m,method,methodopts);
+            freq0=readmatrix("testdata/sinewave-pi6_SCR20_freq.txt");
+            alpha_m0 =readmatrix("testdata/sinewave-pi6_SCR20_alpham.txt");
+            u00=readmatrix("testdata/sinewave-pi6_SCR20_u0.txt");
+            u_fic0=readmatrix("testdata/sinewave-pi6_SCR20_ufic.txt");
+            testCase.verifyTrue(max(abs( ...
+                (freq.data(1:100)-freq0)./freq0))<1e-10,string(opt));
+            testCase.verifyTrue(max(abs( ...
+                (alpha_m(1:100)-alpha_m0)./alpha_m0))<1e-10,string(opt));
+            testCase.verifyTrue(max(abs( ...
+                (u0(1:100)-u00)./u00))<1e-10,string(opt));
+            testCase.verifyTrue(max(abs(( ...
+                u_fic(1:100,idx)-u_fic0(:,idx))./u_fic0(:,idx)))<1e-10, ...
+                string(opt));
+            % 2. opt=randi([2,5]) tests generated according to IECTS
+            opt = randi([2,5],1); idx = randi([1,4],1);
+            if opt==2
+                SCR = 50; fm = 33.3;
+                % TableB.3,fg=60,SCR=50
+                DeltaI_I = [8.040 9.376 11.479 12.844]; % fm=33.3
+            else
+                SCR = 20; fm = 25;
+                % TableB.2,fg=60,SCR=20
+                DeltaI_I = [4.763 5.726 7.640 9.488];% fm=25
+                %DeltaI_I = [3.212 3.958 5.644 7.711];% fm=20
+            end
+            [i_m,u_m]=gen_test_data(Un,In,fg,fs,fm,fv,DeltaI_I,opt,650);
+            if opt==3
+                % B.3.3 Distorted um with multiple zero crossings
+                method = 'stft';methodopts = {...
+                    'Window',rectwin(50000),'OverlapLength',25000,...
+                    'FFTLength',50e3,'FrequencyRange','onesided'};
+            else
+                method = 'ZCD'; methodopts = {};
+            end
+            [~,freq,alpha_m,u0,u_fic] = flicker_ufic(...
+                Sr,Un,SCR,fg,u_m,i_m,method,methodopts);
+            fnm      = ls(sprintf('testdata/B.3.%i*_freq.txt',opt));
```

Comment:
The Matlab code is having a hard time finding the files using this method. I recommend 2 things: 1) decreasing the number of cases you run for the tests and removing unnecessary files from the data folder 2) changing how it is iterating over the files as the code cannot find them.